File: chat-uikit/src/main/java/io/agora/chat/uikit/widget/dialog/DialogViewHelper.java
package io.agora.chat.uikit.widget.dialog;

import android.content.Context;
import android.text.method.LinkMovementMethod;
import android.util.SparseArray;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import java.lang.ref.WeakReference;


class DialogViewHelper {
    private final Context mContext;
    private final View mContentView;
    private SparseArray<WeakReference<View>> views = new SparseArray<>();//Use WeakReference to prevent memory leaks


    public DialogViewHelper(Context mContext) {
        this.mContext = mContext;
        this.mContentView = null;
    }

    public DialogViewHelper(Context mContext, View contentView) {
        this.mContext = mContext;
        this.mContentView = contentView;
    }

    public DialogViewHelper(Context mContext, int contentViewId) {
        this.mContext = mContext;
        this.mContentView = LayoutInflater.from(mContext).inflate(contentViewId, null);
    }

    /**
     * Set click listener
     * @param viewId
     * @param onClickListener
     */
    public void setOnClickListener(int viewId, View.OnClickListener onClickListener) {
        View view = getViewById(viewId);
        if(view != null) {
            view.setOnClickListener(onClickListener);
        }
    }

    <T extends View> T getViewById(int viewId) {
        View view = null;
        WeakReference<View> weakReference = views.get(viewId);
        if (weakReference != null) {
            view = weakReference.get();
        } else {
            if(mContentView != null) {
                view = mContentView.findViewById(viewId);
                views.put(viewId, new WeakReference<View>(view));
            }
        }
        return (T) view;

    }

    /**
     * Set text
     * @param viewId
     * @param text
     */
    public void setText(int viewId, CharSequence text) {
        TextView view = getViewById(viewId);
        if (view != null) {
            view.setText(text);
            //Make hyperlink clickable
            view.setMovementMethod(LinkMovementMethod.getInstance());
        }
    }

    /**
     * @return the dialog's view
     */
    public View getContentView() {
        return mContentView;
    }

    /**
     * Set Image View by resource ID
     * @param viewId
     * @param resId
     */
    public void setImageView(int viewId, int resId) {
        ImageView imageView = getViewById(viewId);
        if(imageView!=null) {
            imageView.setImageResource(resId);
        }
    }
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/common/dialog/DialogViewHelper.kt
package io.agora.uikit.common.dialog

import android.content.Context
import android.text.method.LinkMovementMethod
import android.util.SparseArray
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import java.lang.ref.WeakReference

internal class DialogViewHelper {
    private val mContext: Context

    /**
     * @return the dialog's view
     */
    val contentView: View?
    private val views =
        SparseArray<WeakReference<View?>>() //Use WeakReference to prevent memory leaks

    constructor(mContext: Context) {
        this.mContext = mContext
        contentView = null
    }

    constructor(mContext: Context, contentView: View?) {
        this.mContext = mContext
        this.contentView = contentView
    }

    constructor(mContext: Context, contentViewId: Int) {
        this.mContext = mContext
        contentView = LayoutInflater.from(mContext).inflate(contentViewId, null)
    }

    /**
     * Set click listener
     * @param viewId
     * @param onClickListener
     */
    fun setOnClickListener(viewId: Int, onClickListener: View.OnClickListener?) {
        val view = getViewById<View>(viewId)
        view?.setOnClickListener(onClickListener)
    }

    fun <T : View?> getViewById(viewId: Int): T? {
        var view: View? = null
        val weakReference = views[viewId]
        if (weakReference != null) {
            view = weakReference.get()
        } else {
            if (contentView != null) {
                view = contentView.findViewById(viewId)
                views.put(viewId, WeakReference(view))
            }
        }
        return view as T?
    }

    /**
     * Set text
     * @param viewId
     * @param text
     */
    fun setText(viewId: Int, text: CharSequence?) {
        val view = getViewById<TextView>(viewId)
        if (view != null) {
            view.text = text
            //Make hyperlink clickable
            view.movementMethod = LinkMovementMethod.getInstance()
        }
    }

    /**
     * Set Image View by resource ID
     * @param viewId
     * @param resId
     */
    fun setImageView(viewId: Int, resId: Int) {
        val imageView = getViewById<ImageView>(viewId)
        imageView?.setImageResource(resId)
    }
}